src/eradiate/data/_blind_online.py

```diff
@@ -88,41 +99,63 @@ def fetch(self, filename: PathLike, downloader: t.Callable | None = None) -> Pat
         """
 
         fname = Path(filename).as_posix()
+        url = self.base_url + fname
+        download_attempts = 1 + self.retry_if_failed
+        max_wait = 10
+        result = None
 
         with LoggingContext(
             pooch.get_logger(), level="WARNING"
         ):  # Silence pooch messages temporarily
-            url = self.base_url + fname
-            # Try first to get a compressed file
-            try:
-                return Path(
-                    pooch.retrieve(
-                        url + ".gz",
-                        known_hash=None,
-                        fname=fname + ".gz",
-                        path=self.path,
-                        processor=pooch.processors.Decompress(
-                            name=os.path.basename(fname)
-                        ),
+            for i in range(download_attempts):
+                # Try first to get a compressed file
+                try:
+                    result = Path(
+                        pooch.retrieve(
+                            url + ".gz",
+                            known_hash=None,
+                            fname=fname + ".gz",
+                            path=self.path,
+                            processor=pooch.processors.Decompress(
+                                name=os.path.basename(fname)
+                            ),
+                        )
                     )
-                )
-            except RequestException:
-                pass
+                    break
+                except RequestException:
+                    pass
 
                 # If no gzip-compressed file is available, try the actual file
                 try:
-                    return Path(
+                    result = Path(
                         pooch.retrieve(
                             url,
                             known_hash=None,
                             fname=fname,
                             path=self.path,
                         ),
                     )
-                except RequestException as e:
-                    raise DataError(
-                        f"file '{fname}' could not be retrieved from {self.base_url}"
-                    ) from e
+                    break
+                except RequestException:
+                    pass
+
+                # If we get here, it means download failed
+                retries_left = download_attempts - (i + 1)
+                logger.info(
+                    "Failed to download '%s'. "
+                    "Will attempt the download again %d more time%s.",
+                    fname,
+                    retries_left,
+                    "s" if retries_left > 1 else "",
+                )
+                time.sleep(min(i + 1, max_wait))
```

Comment:
I recommend using a queue of downloads. We would just push the failed download at the end of the queue. A simple dict of attempt numbers can be kept locally and updated when necessary. It would only be the size of the number of failed downloads which in practice is (hopefully) not too big. This could also prepare for an async implementation of the download loop, potentially further improving performances. You could use the size of the failed download counts as a parameter for the time.sleep feature instead, avoiding to reset the delay for each new file.

Reply:
At this point, I don't see a practical way to implement this nicely in store components. As I understand it, the queue should be common to all files; however, store queries are currently on individual files and download tasks are greedy, so I don't think we can fit the queue in the current data store design. I think at this point, the queue system would be implemented by the data fetch program, but I'm not sure if it's really worth the effort since we plan to redesign the data store in a not-so-distant future.

Reply:
Ah yes, I just saw the problem. Hum. The retry mechanism of the data fetch program could indeed be different from the one of the data store. I'm not entirely sure we need both. But let's keep it simple for now. The main objective is achieved